lib/ci/bin/rails_upgrade_analyzer

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env ruby
+
+require "stringio"
+require_relative "../lib/rails_upgrade_analyzer"
+require_relative "../../deprecation_summary"
```

Comment:
Also, it feels strange having a `bin` directory within `lib`.  Can we move this part into the toplevel `bin`?